python/tk_desktop/desktop_window.py

```diff
@@ -365,6 +369,85 @@ def _pop_dll_state(self):
     def contextMenuEvent(self, event):
         self.user_menu.exec_(event.globalPos())
 
+    def _show_rich_message_box(self, icon, title, message, buttons=[]):
+        """
+        Shows a QMessageBox that supports HTML formatting.
+
+        :param icon: Icon to use.
+        :type icon: ``QtGui.QMessageBox.Icon``
+        :param str title: Title of the dialog.
+        :param str message: Message to display.
+        :param list buttons: List of `QtGui.QMessageBox.StandardButton` to display.
+
+        :returns: `QtGui.QMessageBox.StandardButton` value associated with the button
+            that was pressed.
+        """
+        message_box = QtGui.QMessageBox(self)
+        message_box.setIcon(icon)
+        message_box.setTextFormat(QtCore.Qt.RichText)
+        message_box.setWindowTitle(title)
+        message_box.setText(message)
+        for button in buttons:
+            message_box.addButton(button)
+        return message_box.exec_()
+
+    def handle_regen_certs(self):
+        """
+        Regenerates the certificates if the user is certain and restarts the Shotgun Desktop on
+        demand.
+        """
+
+        # Need to create the message box by hand to have rich text format, hence
+        # clickable Urls.
+        choice = self._show_rich_message_box(
+            QtGui.QMessageBox.Information,
+            "Shotgun browser integration",
+            "Regenerating the Shotgun Desktop's browser integration certificates should "
+            "only be done if you have issues with the browser integration.<br/>"
+            "<br/>"
+            "If you are unsure how to proceed, we recommend you visit our support page "
+            "to troubleshoot connections with <a href='%s'>Chrome</a> or "
+            "<a href='%s'>Firefox</a>.<br/>"
+            "<br/>"
+            "Would you like to continue?" % (
+                "https://support.shotgunsoftware.com/hc/en-us/articles/114094536273",
+                "https://support.shotgunsoftware.com/hc/en-us/articles/115000054954"
+            ),
+            [QtGui.QMessageBox.Yes, QtGui.QMessageBox.No]
+        )
+
+        if choice != QtGui.QMessageBox.Yes:
+            return
+
+        try:
+            desktop_server_framework.regenerate_certificates(self)
+        except Exception:
+            log.exception("Unexpected error while regenerating certificates:")
+            self._show_rich_message_box(
+                QtGui.QMessageBox.Critical,
+                "Shotgun browser integration",
+                "It appears there are an issue while regenerating the certificates."
+                "\n"
+                "Please contact <a href='{0}'>our support team</a> "
+                "if you need assistance resolving this issue. Make sure to attach the logs found "
+                "at <a href='file://{1}'>{1}</a>.".format(
+                    "mailto:support@shotgunsoftware.com",
+                    sgtk.LogManager().base_file_handler.baseFilename
```

Comment:
how about instead recommending them to zip up their entire log folder and send that? The code will be simpler but more importantly i have recently found that clients sometimes just send snippets of their log and you miss the big picture. by asking them to send all logs you get everything and there is no risk of confusion.